### ISSUE TITLE
Formatting lost when using subdoc()

### ISSUE THREAD

**Issue by oussama-he** [elapouya/python-docx-template#388]
I have a DOCX template and I want to insert a sub document into it using the `{{p }}` tag, it works fine for now except that some formattings are lost. In the original document (sub_doc.docx), the title is italic, centered, the size of the main font in the table 10, in the result file the title lost the italic and centered formatting.

Pleas how can I keep the formatting when inserting the sub document?

```python
from docxtpl import DocxTemplate
tpl = DocxTemplate('tpl.docx')
sd = tpl.new_subdoc('sub_doc.docx')
context = {
'sd' : sd,
}
tpl.render(context)
tpl.save('result.docx')
```

**Comment by elapouya** [comment]
Please, provide a fully runnable test case including your docx's

**Comment by romdespoel** [comment]
Finding the same issue here. I'm attaching the master/sub doc. I was using the script you provide in the repo's tests:

```
from docxtpl import DocxTemplate

tpl = DocxTemplate('merge_docx_master_tpl.docx')
sd = tpl.new_subdoc('merge_docx_subdoc.docx')

context = {
    'mysubdoc': sd,
    'var':3,
}

tpl.render(context)
tpl.save('merge_docx.docx')
```

[merge_docx_subdoc.docx](https://github.com/elapouya/python-docx-template/files/7455386/merge_docx_subdoc.docx)
[merge_docx_master_tpl.docx](https://github.com/elapouya/python-docx-template/files/7455387/merge_docx_master_tpl.docx)

Additionally, tags in the subdoc don't get replaced in the render but this is not a big deal as you can render the subdoc first.

Let me know if I can help by providing anything else

**Comment by elapouya** [comment]
I confirm some formatting are lost (italic, bold in my case), investigating...

**Comment by elapouya** [comment]
Bad, manipulation, formatting is conserved with latest docxtpl version (italic, bold in my case with @romdespoel test case)
But subdoc jinja2 tags are not rendered.

**Comment by elapouya** [comment]
I looked at the code : to have subdoc tags rendered, one would have to specify a specific context to subdoc object then I have to modify the code to have subdoc rendered internally. This is almost the same thing as doing it separately. So, I will not code that.
To summary, if you want some jinja2 tags to be rendered in a subdoc : do not expect to be done in a single pass when rendering the master document, but instead : 
1) render the subdoc with docxtpl and a specific context
2) render the master doc with docxtpl with a context containing the subdoc object.

Closing this issue.
